src/containers/BookForm.js

```diff
@@ -1,19 +1,44 @@
-import React from 'react';
+import React, { useState } from 'react';
+import { useDispatch } from 'react-redux';
+import shortid from 'shortid';
+import { CREATE_BOOK } from '../actions/index';
 
-const categories = ['Action', 'Biography', 'History', 'Horror', 'Kids', 'Learning', 'Sci-Fi'];
+const categories = ['', 'Action', 'Biography', 'History', 'Horror', 'Kids', 'Learning', 'Sci-Fi'];
 
 const BookForm = () => {
   const category = categories.map(category => (
     <option key={category} value={category}>{ category }</option>
   ));
+  const [inputValue, setInputValue] = useState('');
+  const [selectValue, setSelectValue] = useState('');
+
+  const handleInputChange = e => {
+    setInputValue(e.target.value);
+  };
+  const handleSelectChange = e => {
+    setSelectValue(e.target.value);
+  };
+  const dispatch = useDispatch();
+
   return (
     <>
       <form>
-        <input type="text" />
-        <select>
+        <input value={inputValue} onChange={handleInputChange} type="text" />
+        <select value={selectValue} onChange={handleSelectChange}>
           {category}
         </select>
-        <input type="submit" />
+        <button
+          onClick={() => {
+            dispatch(CREATE_BOOK({
+              id: shortid.generate(),
+              title: inputValue,
+              category: selectValue,
+            }));
+          }}
```

Comment:
Implements `handleSubmit` and attaches it to form's onSubmit event ([screenshot from pathwright](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/react_reducx/bookstore/milestone3/HandleSubmit.png)).
Let's create a separate event handler method and attach it to the button onClick attribute. After dispatching the action, the state should be reset to handle another form submission.